examples/direct-upload/README.md

```diff
@@ -438,6 +448,181 @@ server.start((err) => {
 ```
+
 #### We now have a server that can our index.html can communicate with!
 
-### Step 5 - Write the client side code to send our requests to the backend and to S3
+### Step 5 - Write the client side code to send our requests to the backend and then to S3
+
++ Create a `public` directory in the root of your project. Inside this new folder
+create two new files `index.html` and `client.js`
+
+Insert the following into your `index.html`:
+
+```html
+<!DOCTYPE html>
+<html>
+  <head>
+    <title>S3 Upload Demo</title>
+    // optional stylesheet
+    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/flat-ui/2.2.2/css/flat-ui.min.css">
+    // include your client.js file that we are going to write
+    <script type="text/javascript" src="client.js"></script>
+  </head>
+  <body>
+    <h1>S3 Direct Upload Demo</h1>
+    // simple form with a file input (we'll be adding more later dynamically)
+    <form>
+    // onchange gets fired when a file is selected so we want to save the file
+      <input id="fileInput" type="file" name="file" onchange="uploadDemo.saveFile(this.files)"/>
+    </form>
+    // button that will submit our file
+    <button onclick="uploadDemo.submitFile()">Submit</button>
+    // container for the success message and link to our image
+    <div class="successMessageContainer">
+      <a class="imageLink"></a>
+    </div>
+  </body>
+</html>
+```
+In your `client.js` file add the following:
+
+```js
+// wrap everything in an IIFE (immediately invoked function expression) to contain
+// global variables
+var uploadDemo = (function () {
+  // 'global' variable used to store our filename
+  var filename
+
+  /**
+  * Saves the filename to our global variable when a file has been selected
+  * @param {Object} file - file from our file input tag
+  **/
+  function saveFile (file) {
+    filename = file[0].name
+  }
+
+  /**
+  * Calls our getCredentialsFromServer function with the global filename
+  **/
+  function submitFile () {
+    getCredentialsFromServer(filename)
+  }
+
+  // function that retrieves our S3 credentials from our server
+  /**
+  * Saves the filename to our global variable when a file has been selected
+  * @param {string} filename - name of the file we want to upload
+  **/
+  function getCredentialsFromServer (filename) {
+    var xhttp = new XMLHttpRequest()
+    xhttp.onreadystatechange = function () {
+      if (this.readyState === 4 && this.status === 200) {
+        // after we've received a response we want to assign it to a variable
+        var s3Data = JSON.parse(xhttp.responseText)
+        // call our buildAndSubmitForm function
+        buildAndSubmitForm(s3Data)
+        // return a success message after the image has been uploaded along with
+        // link to image
+        var successMessage = document.createElement('h4')
+        successMessage.innerHTML = 'Image Successfully Uploaded at: '
+        var link = `https://<your_bucket_name>.s3.amazonaws.com/${filename}`
```

Comment:
could the `AWS_S3_BUCKET_NAME` be an _environment variable_ returned by the _server_ in the `getCredentialsFromServer` call?